src/commands/ban.js

```diff
@@ -2,9 +2,42 @@ const Discord = require("discord.js");
 const errors = require("../utils/errors.js");
 
 module.exports.run = async (bot, message, args) => {
-//stub
+
+    if(!message.member.hasPermission('MANAGE_MESSAGES')){ //check permissions
+        return errors.noPerms(message, "Moderator Role Needed")
+    }
+
+    args = message.content.split(' ').slice(1);
+    const userToBan = message.mentions.users.first();
+    const banReason = args.slice(1).join(' ');
+    
+    if (userToBan) {
+        if (userToBan.length > 2) {
+            return message.channel.send(`${message.author} Invalid format. Please ban one user at a time.`);
+        }
+        else {
+            if (message.author === userToBan) {
+                return message.channel.send(`${message.author} You cannot ban yourself.`);
+            }
+            if (!banReason) {
```

Comment:
Let's create a default ban reason rather than forcing them to specify a reason to ban.

Reply:
will do